libafl_frida/src/coverage_rt.rs

```diff
@@ -61,128 +142,318 @@ impl CoverageRuntime {
     /// Create a new coverage runtime
     #[must_use]
     pub fn new() -> Self {
-        Self(Rc::pin(RefCell::new(CoverageRuntimeInner {
-            map: [0_u8; MAP_SIZE],
-            previous_pc: 0,
-            _pinned: PhantomPinned,
-        })))
+        Self {
+            inner: Rc::pin(RefCell::new(CoverageRuntimeInner {
+                map: [0_u8; MAP_SIZE],
+                previous_pc: 0,
+                _pinned: PhantomPinned,
+            })),
+            inner_bbs: Rc::pin(RefCell::new(CoverageRuntimeInner {
+                map: [0_u8; MAP_SIZE],
+                previous_pc: 0,
+                _pinned: PhantomPinned,
+            })),
+            ranges: RangeMap::new(),
+            save_dr_cov: false,
+            coverage_directory: PathBuf::from("./coverage"),
+            basic_blocks: HashMap::new(),
+            cnt: 0,
+            max_cnt: 0,
+            stored_cnt: 0,
+        }
+    }
+
+    /// Set whether to save `DrCov` files
+    #[must_use]
+    pub fn save_dr_cov(mut self, save_dr_cov: bool) -> Self {
+        self.save_dr_cov = save_dr_cov;
+        self
+    }
+
+    /// Set the coverage directory
+    #[must_use]
+    pub fn coverage_directory(mut self, coverage_directory: &str) -> Self {
+        self.coverage_directory = PathBuf::from(coverage_directory);
+        self
+    }
+
+    /// Set the maximum number of executions to accumulate before writing the coverage to disk
+    #[must_use]
+    pub fn max_cnt(mut self, max_cnt: usize) -> Self {
+        self.max_cnt = max_cnt;
+        self
+    }
+
+    /// Builder pattern
+    #[must_use]
+    pub fn build(self) -> CoverageRuntime {
```

Comment:
Now that the CoverageRuntime is a builder, you need to rename it to reflect that.

Reply:
I did not see much value in adding more classes and code that do nothing useful, so I changed it back to new().